### PR TITLE
@task PDA-1220 GET /v1/locations/location - build graphQL query.

### PR DESCRIPTION
**Pull request by awang-parasoft**

### CODE REVIEW

**Comment by yshi-parasoft** [comment]
image
There might be some issue with the deserialization. If I try to select "region" in the result it always return null.

File: build.gradle [hunk 1]
```diff
@@ -54,6 +54,10 @@ dependencies {
 	providedRuntime 'org.springframework.boot:spring-boot-starter-tomcat'
 	testImplementation 'org.springframework.boot:spring-boot-starter-test'
 	testImplementation 'org.springframework.security:spring-security-test'
+	testImplementation 'com.graphql-java-kickstart:graphql-spring-boot-test:14.0.0'
+	testImplementation 'com.graphql-java-kickstart:graphql-java-kickstart:14.0.0'
```
Comment (yshi-parasoft) [inline build.gradle:58]: Do we need this dependency?

File: src/test/resources/graphql/locations/location.graphql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+query LocationInfo($region: RegionType!) {
```
Comment (yshi-parasoft) [inline src/test/resources/graphql/locations/location.graphql:1]: Maybe rename this file to "getLocation".

**Comment by awang-parasoft** [comment]
> "region" field appeared in selections and if I select it, it will always return null which caused failure as below because of "not null" rule in the schema: image
> 
> Even if I remove the "not null" rule, I'll get it as null which looks weird. image
> 
> So we should either handle the deserialization issue OR if we don't need that field anywhere please just hide it.

REST API doesn't return the region, maybe we can remove the region in the Location schema type.

File: build.gradle (hunk 1 above)
Comment (awang-parasoft) [reply build.gradle:58]: This is used in the test.

File: src/test/resources/graphql/locations/location.graphql (hunk 2 above)
Comment (awang-parasoft) [reply src/test/resources/graphql/locations/location.graphql:1]: OK

**Pull request merged**